src/utils/generateTheme.js

```diff
@@ -0,0 +1,12 @@
+import generateColors from '../utils/generateColors';
+import extractRoleFromPath from './extractRoleFromPath';
+
+function generateTheme(currentLocation) {
+  const roleAsColorVariant = extractRoleFromPath(currentLocation.toLowerCase());
```

Comment:
You are extracting `role` from the `path`. So name the variable just `role`. Also perform the lower casing inside `extractRoleFromPath` method.